@nowarn breaks forwards compatibility at compile-time

```
$> cat A.scala
trait A {
  @annotation.nowarn val x = 0
}

$> scv 2.13.2 A.scala

$> cat B.scala
class B extends A

$> scv 2.13.1 B.scala
B.scala:1: error: Symbol 'type scala.annotation.nowarn' is missing from the classpath.
This symbol is required by ' <none>'.
Make sure that type nowarn is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
A full rebuild may help if 'A.class' was compiled against an incompatible version of scala.annotation.
class B extends A
                ^
one error found
```

--- a/src/reflect/scala/reflect/internal/Symbols.scala
+++ b/src/reflect/scala/reflect/internal/Symbols.scala
@@ -124,7 +124,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     def isJavaEnum: Boolean = hasJavaEnumFlag
     def isJavaAnnotation: Boolean = hasJavaAnnotationFlag
     def isStaticAnnotation: Boolean =
-      hasJavaAnnotationFlag || isNonBottomSubClass(StaticAnnotationClass)
+      hasJavaAnnotationFlag || isNonBottomSubClass(StaticAnnotationClass) && this != NowarnClass
 
     def newNestedSymbol(name: Name, pos: Position, newFlags: Long, isClass: Boolean): Symbol = name match {
       case n: TermName => newTermSymbol(n, pos, newFlags)

